### ISSUE TITLE
项目自荐 | 语义分割经典网络模型 PyTorch 实现

### ISSUE THREAD

**Issue by Charmve** [521xueweihan/HelloGitHub#1683]
## 项目推荐
- 项目地址：https://github.com/Charmve/Semantic-Segmentation-PyTorch

- 类别：Python、机器学习
- 目前已经完成的网络、模型
1. Vanilla FCN: FCN32, FCN16, FCN8, in the versions of VGG, ResNet and DenseNet respectively
([Fully convolutional networks for semantic segmentation](http://www.cv-foundation.org/openaccess/content_cvpr_2015/papers/Long_Fully_Convolutional_Networks_2015_CVPR_paper.pdf))
2. U-Net ([U-net: Convolutional networks for biomedical image segmentation](https://arxiv.org/pdf/1505.04597))
3. SegNet ([Segnet: A deep convolutional encoder-decoder architecture for image segmentation](https://arxiv.org/pdf/1511.00561))
4. PSPNet ([Pyramid scene parsing network](https://arxiv.org/pdf/1612.01105))
5. GCN ([Large Kernel Matters](https://arxiv.org/pdf/1703.02719))
6. DUC, HDC ([understanding convolution for semantic segmentation](https://arxiv.org/pdf/1702.08502.pdf))
7. Mask-RCNN ([paper](https://arxiv.org/abs/1703.06870), [code from FAIR](https://github.com/facebookresearch/Detectron), [<b>code PyTorch</b>](https://github.com/multimodallearning/pytorch-mask-rcnn))

- 项目后续更新计划：

我将首先在PyTorch中实现<a href="https://github.com/Charmve/PaperWeeklyAI/tree/master/05_Image%20Segmentation" target="_blank"><i>The Image Segmentation Paper Top10 Net</i></a>。

- [ ] DeepLab v3  
- [ ] RefineNet 
- [ ] ImageNet
- [ ] GoogleNet
- [ ] More dataset (e.g. ADE)

- 项目描述：PyTorch implementation for Semantic Segmentation, include FCN, U-Net, SegNet, GCN, PSPNet, Deeplabv3, Deeplabv3+, Mask R-CNN, DUC, GoogleNet, and more dataset

- 推荐理由：最简单实用的语义分割源代码！对初学者友好，对工程实践者便利！

- 示例代码：（可选）长度：1-20 行

U-Net两步网络的实现

<pre>
<code>
class _EncoderBlock(nn.Module):
    def __init__(self, in_channels, out_channels, dropout=False):
        super(_EncoderBlock, self).__init__()
        layers = [
            nn.Conv2d(in_channels, out_channels, kernel_size=3),
            nn.BatchNorm2d(out_channels),
            nn.ReLU(inplace=True),
            nn.Conv2d(out_channels, out_channels, kernel_size=3),
            nn.BatchNorm2d(out_channels),
            nn.ReLU(inplace=True),
        ]
        if dropout:
            layers.append(nn.Dropout())
        layers.append(nn.MaxPool2d(kernel_size=2, stride=2))
        self.encode = nn.Sequential(*layers)

    def forward(self, x):
        return self.encode(x)

class _DecoderBlock(nn.Module):
    def __init__(self, in_channels, middle_channels, out_channels):
        super(_DecoderBlock, self).__init__()
        self.decode = nn.Sequential(
            nn.Conv2d(in_channels, middle_channels, kernel_size=3),
            nn.BatchNorm2d(middle_channels),
            nn.ReLU(inplace=True),
            nn.Conv2d(middle_channels, middle_channels, kernel_size=3),
            nn.BatchNorm2d(middle_channels),
            nn.ReLU(inplace=True),
            nn.ConvTranspose2d(middle_channels, out_channels, kernel_size=2, stride=2),
        )
</pre>
</code>

- 截图：（可选）gif/png/jpg

![image](https://user-images.githubusercontent.com/29084184/112089925-858b7d80-8bcd-11eb-9465-9e7809c36bae.png)

**Comment by 521xueweihan** [comment]
非常感谢您推荐项目。

**该项目暂不能收录到 HelloGitHub 月刊中**，HelloGitHub 推荐项目审核标准 #271。
期待持续完善该项目，后续推荐更多的项目。

再次感谢您对 HelloGitHub 的支持 🙏
